gr-blocks/lib/complex_to_interleaved_char_impl.cc

```diff
@@ -37,14 +42,11 @@ int complex_to_interleaved_char_impl::work(int noutput_items,
                                            gr_vector_const_void_star& input_items,
                                            gr_vector_void_star& output_items)
 {
-    const gr_complex* in = (const gr_complex*)input_items[0];
-    char* out = (char*)output_items[0];
-
-    int npairs = (d_vector ? noutput_items : noutput_items / 2);
-    for (int i = 0; i < npairs; i++) {
-        *out++ = (char)lrintf(in[i].real()); // FIXME saturate?
-        *out++ = (char)lrintf(in[i].imag());
-    }
+    const float* in = (const float*)input_items[0];
+    int8_t* out = (int8_t*)output_items[0];
+
+    volk_32f_s32f_convert_8i(
+        out, in, d_scalar, d_vector ? noutput_items * 2 : noutput_items);
```

Comment:
Thinking through this: If a vector of length 2x sizeof(char), noutput_items will be in # of vectors, so it needs 2x to get the correct number of total elements to manipulate. If a vector of length 1x, then the data will in in I/Q pairs, so you'll need 2x noutput_items for that option. In other words, either way you need 2x noutput_items ... so, please fix that & add something like what I wrote here as a comment for future reference. Here and for `short` too.

Reply:
Or maybe I'm wrong? been a long day ...

Reply:
Ping on my thinking here. I think I'm correct for the "complex to *" blocks. I'll comment on the "* to complex" blocks in another comment.

Reply:
Same logic here as I just noted for the "X to complex" block, just in reverse: 1 input item will generate either 1 vector item output or 2 non-vector items output. Either way, for Volk purposes, we're processing 2 items. Hence, we always need to process `2 * noutput_items` items in Volk.

Reply:
Logic on these as an interpolator goes that noutput_items with an interp of 2 will be 2x the number of complex's in, but will match1-to-1 the number of floats in.  So it should be a direct map (noutput_items).  If it's a vector in and there's twice as much in then just 2*noutput_items.

Reply:
OK. This is correct. I was messing up `noutput_items` values. On to the other way.